lib/routes/sms.js

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const error = require('../error')
+const libpn = require('google-libphonenumber')
+const validators = require('./validators')
+
+const METRICS_CONTEXT_SCHEMA = require('../metrics/context').schema
+const SENDER_IDS = new Map(require('../sms-sender-ids.json'))
+
+module.exports = (log, isA, error, customs, sms) => {
+  return [
+    {
+      method: 'POST',
+      path: '/sms',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            phoneNumber: isA.string().regex(validators.E164_NUMBER).required(),
+            messageId: isA.number().positive().required(),
+            metricsContext: METRICS_CONTEXT_SCHEMA
+          }
+        }
+      },
+      handler (request, reply) {
+        log.begin('sms.send', request)
+        request.validateMetricsContext()
+
+        const sessionToken = request.auth.credentials
+        const phoneNumber = request.payload.phoneNumber
+        const messageId = request.payload.messageId
+        const acceptLanguage = request.app.acceptLanguage
+
+        customs.check(request, sessionToken.email, 'connectDeviceSms')
+          .then(() => getSenderId(phoneNumber))
+          .then(senderId => {
+            return sms.send(phoneNumber, senderId, messageId, acceptLanguage)
+              .then(() => request.emitMetricsEvent(`sms.${messageId}.sent`))
+              .catch(err => {
+                if (err.status === 500) {
+                  throw error.messageRejected(err.reason, err.reasonCode)
```

Comment:
> I could parse it to a number but it seems wrong and I'm not sure there's any point?

I'll do it on the front end then. The reasonCode is used to convert to an internal error with an errno, where the reasonCode becomes the errno. Since we use strict equality operators when looking up errors by errno, strings don't work.